include/extractor/scripting_environment.hpp

```diff
@@ -65,8 +77,14 @@ class ScriptingEnvironment
                     tbb::concurrent_vector<std::pair<std::size_t, ExtractionWay>> &resulting_ways,
                     tbb::concurrent_vector<boost::optional<InputRestrictionContainer>>
                         &resulting_restrictions) = 0;
+
+    static location_handler_sparse_type *location_sparse_handler;
+    static location_handler_dense_type *location_dense_handler;
+    static void setSparseCache(location_handler_sparse_type &cache) {
+        location_sparse_handler = &cache;};
+    static void setDenseCache(location_handler_dense_type &cache) {
+        location_dense_handler = &cache;};
```

Comment:
Cache is static because i can't access into luabind anonymous function ?
```
.def("location", +[](const osmium::NodeRef& nref){
                 if(location_sparse_handler)
                 {
                     osmium::Location location = LuaScriptingEnvironment::location_sparse_handler->get_node_location(nref.ref());
                     return location;
                 }
                 else if(location_sparse_handler)
                 {
                     osmium::Location location = LuaScriptingEnvironment::location_sparse_handler->get_node_location(nref.ref());
                     return location;
                 }
                 else'
                     return nref.location();
                     return location;
                 }
                 else
                     return nref.location();
})
```